tensorflow/c/experimental/filesystem/filesystem_interface.h

```diff
@@ -83,23 +83,25 @@ typedef struct TF_TransactionToken {
   TF_Filesystem* owner;
 } TF_TransactionToken;
 
-typedef struct TF_Filesystem_Option_Value {
-  int type_tag;
-  int num_values;
-  union {
-    int64_t inv_val;
-    double real_val;
-    struct {
-      char* buf;
-      int buf_length;
-    } buffer_val;
-  } * values;  // owned
+typedef union TF_Filesystem_Option_Value {
+  int64_t inv_val;
+  double real_val;
+  struct {
+    char* buf;
+    int buf_length;
+  } buffer_val;
 } TF_Filesystem_Option_Value;
 
+constexpr int TF_FILESYSTEM_OPTION_TYPE_INT = 0;
+constexpr int TF_FILESYSTEM_OPTION_TYPE_REAL = 1;
+constexpr int TF_FILESYSTEM_OPTION_TYPE_BUFFER = 2;
```

Comment:
Would be nice to make these into an enum.

Reply:
@mihaimaruseac Done.